queries/modules/io/28/modules/xbrl/components.jq

```
@@ -91,6 +111,33 @@ declare function components:components($component-or-ids as item*) as object*
     )
 };
 
+declare function components:components-for(
+  $archives-or-ids as item*,
+  $roles as string*,
+  $concepts as string*,
+  $exact-labels as string*,
+  $options as object*
+) as object*
+{
+  let $aids := if(deep-equal($archives-or-ids, $components:ALL-ARCHIVES))
+               then $components:ALL-ARCHIVES
+               else archives:aid($archives-or-ids)
+  let $query := {|
```

could we throw an error if all of the parameters are set to $components:ALL-*?